Lesson4

JustSenoko opened the pull request:
попробовала сделать и fxml, и swing

JustSenoko committed:
окно чата на fxml

JustSenoko committed:
fxml:
вынесла общие классы в корневую папку урока
немного оптимизировала код

JustSenoko committed:
окно чата на swing

JustSenoko committed:
переименовала поля

usharik commented on src/ru/geekbrains/lesson4/swing/MainWindow.java:
```diff
@@ -0,0 +1,44 @@
+                JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        add(scroll, BorderLayout.CENTER);
+
+        JPanel sendMessagePanel = new JPanel();
```
Контролы лучше объявлять как поля класса-окна которому они принадлежат. Это общепринятый подход для такого рода классов.

usharik commented on src/ru/geekbrains/lesson4/swing/ChatActionListener.java:
```diff
@@ -0,0 +1,32 @@
+    private Chat chat;
+    private Participant participant;
+
+    ChatActionListener(JTextField msgField, JTextArea chatArea, Chat chat, Participant participant) {
```
Листнеры чаще всего объявляют в виде анонимных классов. Пожалуйста переделайте перед тем как начнете писать сетевой чат. Такая структура кода создаст вам массу неудобств.

JustSenoko committed:
swing:
вынесла контролы в поля класса MainWindow
переделала ActionListener, чтобы не вводить отдельный класс
переписала элемент список чата, чтобы не нужен был отдельный класс Chat
переименовала Participant в ChatUser
